### ISSUE TITLE
should_use_colors inside debugger

### ISSUE THREAD

**Issue by euri10** [Kludex/uvicorn#497]
the colorized formatter in pycharm debugger doesnt work (it displays everything in black) because for the `AccessFormatter` we have
```
    def should_use_colors(self):
        return sys.stdout.isatty()
```
that returns False and same for the `DefaultFormatter`, where
```
    def should_use_colors(self):
        return sys.stderr.isatty()
``` 
that returns the same False

In that particular case, the process is ran with [pydevd](https://github.com/fabioz/PyDev.Debugger).
I'm not familar with its internals, but a quick look makes me think it redirects stderr/stdout 
https://github.com/fabioz/PyDev.Debugger/search?q=isatty&type=Code 

do you think the `should_use_colors` test could be improved or that I should raise an issue there ?

**Comment by lovelydinosaur** [comment]
> do you think the should_use_colors test could be improved

I'm not aware that there's anything smarter we can do there?

What do we think about adding a `--log-colors`/`--no-log-colors` command line flag, that explicitly forces the colors on/off, rather than relying on auto-detection?

Any pull requests for that would be very welcome.

**Comment by lovelydinosaur** [comment]
Closed via #502
